performant-software/dm-2
# Text editor bugfixes, and strikethroughs

### What this PR does
- Per #201:
  - Adds "CMD-U to underline" functionality
- Per #202. #84:
  - Keeps text selection visible when clicking on the toolbar (by using `e.preventDefault` and `mouseDown` instead of `click` events)
- Per #289:
  - Adds strikethrough functionality (no key command yet)
- Bumps all ProseMirror modules to latest non-breaking versions
  - Adds `resolutions` block to `package.json` to resolve redundant deps

### Additional notes

Though this PR adds strikethroughs, I did not link it to #289 because that issue should remain open until all features are implemented.

### Status


### Steps to test

1. Visit https://dm-2-staging.herokuapp.com/
2. Log in
3. Open or create a project with Write or Admin access
4. Open and check out or create a text document

Issue #201:

5. Select some text, press CMD-U (CTRL-U on PC), and verify that it becomes underlined

Issue #202:

6. Select some text, click on various things on the toolbar, and verify that the selection does not disappear

Issue #289:

7. Select some text, use the strikethrough icon (to the right of underline) on the toolbar, and verify that it gets a <s>strikethrough</s> applied
